src/utils/database-common/src/transactions/db_transaction_manager.rs

```diff
@@ -34,14 +34,26 @@ pub trait DatabaseTransactionManager: Send + Sync {
 
 /////////////////////////////////////////////////////////////////////////////////////////
 
-pub async fn run_transactional<H, HFut>(
-    base_catalog: &Catalog,
+/// Enriching passed catalogs with [`TransactionRef`]
+///
+/// # Arguments
+///
+/// * `catalogs` - Catalogs in which `base_catalog` is expected to be the first
+///   one
+/// * `callback` - A callback that receives updated catalogs. If necessary, we
+///   can return a value from it
+pub async fn run_transactional<H, HFut, HFutResultT, const N: usize>(
```

Comment:
I totally don't get the idea what you are trying to do here, and why do we need to send multiple catalog ???